core/src/executor/evaluate/function.rs

```diff
@@ -79,6 +79,27 @@ pub fn concat(exprs: Vec<Evaluated<'_>>) -> Result<Evaluated> {
     }
 }
 
+pub fn concat_ws<'a>(
+    name: String,
+    separator: Evaluated<'a>,
+    exprs: Vec<Evaluated<'a>>,
+) -> Result<Evaluated<'a>> {
+    let separator = eval_to_str!(name, separator);
+
+    let result = exprs
+        .into_iter()
+        .map(|expr| expr.try_into())
+        .filter(|value| !matches!(value, Ok(Value::Null)))
+        .map(|value| match value {
+            Ok(value) => Ok(String::from(value)),
+            Err(_err) => Ok(String::from(Value::Str("".to_owned()))),
```

Comment:
It would be good to consider errors as terminal nodes.
Other states can be converted into error, but no case from error to non-error cases.

Though we know that it is currently not possible to write tests which reaches to this, it is a hidden context which others cannot recognize.

It is ok that we don't have a test case for this, but it can be a problem if we ignore the error because we cannot track the changes from outside.